ECMWF IFS change

alcoat opened the pull request:
From : https://www.ecmwf.int/en/about/media-centre/news/2024/ecmwf-releases-much-larger-open-dataset
"Medium-range data of the Integrated Forecasting System (IFS) are provided at a resolution of 0.25 x 0.25 degrees (28 x 28 km), compared to 0.4 x 0.4 degrees before, and some additional parameters will be added."

So update of file paths

blaylockbk commented:
(This will need updates to the docs and tutorial notebooks, but I can handle that after we get the new templates merged.)

blaylockbk commented:
Hi @alcoat, 

First, please forgive me for not being too experienced working on pull requests with others; I'm still figuring this out.

I made some additional changes based on your. Take a look and see what you think about this approach: https://github.com/alcoat/Herbie/compare/main...blaylockbk:Herbie:update_ecmwf_ifs

I want to make sure you get credit for this pull request, so I think if you like some of my changes, you can pull them into your branch??

blaylockbk commented:
Whoo-hoo! Herbie found the data from my additional changes
![image](https://github.com/blaylockbk/Herbie/assets/6249613/3ccbf209-86ba-400b-8c1f-430d9db57a68)

blaylockbk commented:
hold on. One thing I didn't think about is accessing _historical_ data (e.g., when a user asks for data from January 2024); that still needs to be handled. Perhaps an if condition by the date??

alcoat commented:
Hi @blaylockbk 
I've updated the ecwmf dataset in order to access archives for ifs model: two dates are used, one for the update of the 0p25 vs 0p4 resolution and the other one the the ifs and aifs directories. So I've added some tests which are testing the old datasets

blaylockbk commented:
This is awesome! Thanks.

blaylockbk commented:
FYI, I'll work on a Herbie release tonight or tomorrow.
